rust-toolchain

```diff
@@ -1,3 +1,4 @@
 [toolchain]
 channel = "nightly-2021-01-31"
 components = ["rustfmt", "clippy"]
+targets = [ "x86_64-unknown-linux-gnu", "armv7-unknown-linux-gnueabihf", "x86_64-apple-darwin", "x86_64-pc-windows-msvc"]
```

Comment:
Do we really need all targets here? That is gonna install all of them on all machines no?

Reply:
I remember we discussed this in the context of the error "target not installed" on CI and you mentioned that I could solve that by defining the targets in the toolchain file. Is there a better way?

Reply:
I think you only need to add the non-obvious target `armv7-unknown-linux-gnueabihf`. All others are installed by default if the current OS requires it. In other words, on Linux, `x86_64-unknown-linux-gnu` will be installed anyway because that is the current host OS and is what the binary will be built against.

However, if you add all of them here, rustup is going to install the apple and windows targets on a Linux machine as well even though we are never going to cross-compile to them and it is not even that easy I believe.

In summary, I think the only the thing you need to add here is the `armv7-unknown-linux-gnueabihf` target and you should be good to go to cross-compile to it from the `ubuntu-latest` runner in CI.

Reply:
I see, thanks for the clarification!